### PR TITLE
fix disk related problems

### PR DESCRIPTION
**Pull request by SlavaValAl**

### CODE REVIEW

File: octane/patches/update_release_partition_info.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# under the License.
+
+from nailgun.db import db
+from nailgun.db.sqlalchemy import models
+
+releases = db().query(models.Release)
+for rel in releases:
+    meta = rel.volumes_metadata
+    volumes = meta['volumes']
+    for v_id in xrange(len(volumes)):
```
Comment (YorikSar) [inline octane/patches/update_release_partition_info.py]: you meant `for volume in volumes`, right?

File: octane/patches/update_release_partition_info.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        volume = volumes[v_id]
+        if volume['min_size']['generator'] == 'calc_min_log_size':
+            volumes[v_id]['min_size']['generator'] = 'calc_gb_to_mb'
+            volumes[v_id]['min_size']['generator_args'] = [2]
+    db().query(models.Release).filter_by(id=rel.id).update(
+        {"volumes_metadata": meta})
+
+db().commit()
+db().flush()
```
Comment (YorikSar) [inline octane/patches/update_release_partition_info.py]: No need for flush here.

File: octane/patches/update_release_partition_info.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from nailgun.db import db
+from nailgun.db.sqlalchemy import models
+
+releases = db().query(models.Release)
```
Comment (YorikSar) [inline octane/patches/update_release_partition_info.py]: You should start a transaction before this.

Comment (SlavaValAl) [inline octane/patches/update_release_partition_info.py]: https://github.com/stackforge/fuel-web/blob/master/nailgun/nailgun/db/sqlalchemy/__init__.py#L80-L88
https://github.com/stackforge/fuel-web/blob/master/nailgun/nailgun/objects/node.py#L145-L146

It's can be used as it is

File: octane/patches/update_release_partition_info.py (hunk 1 above)
Comment (SlavaValAl) [inline octane/patches/update_release_partition_info.py]: It's only way I know how to modify list when I'm iterating over it.

Comment (YorikSar) [inline octane/patches/update_release_partition_info.py]: You aren't changing list. You're only changing objects stored in it.

File: octane/patches/update_release_partition_info.py (hunk 2 above)
Comment (SlavaValAl) [inline octane/patches/update_release_partition_info.py]: fixed

File: octane/patches/update_release_partition_info.py (hunk 1 above)
Comment (SlavaValAl) [inline octane/patches/update_release_partition_info.py]: fixed

File: octane/patches/update_release_partition_info.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+for rel in releases:
+    meta = rel.volumes_metadata
+    volumes = meta['volumes']
+    for v_id in xrange(len(volumes)):
+        volume = volumes[v_id]
+        if volume['min_size']['generator'] == 'calc_min_log_size':
+            volumes[v_id]['min_size']['generator'] = 'calc_gb_to_mb'
+            volumes[v_id]['min_size']['generator_args'] = [2]
+    db().query(models.Release).filter_by(id=rel.id).update(
+        {"volumes_metadata": meta})
```
Comment (SlavaValAl) [inline octane/patches/update_release_partition_info.py]: JSON fields are immutable from sqlalchemy point of view...that's why i fixed this issue in this was...another approach(to make JSON mutable) will require more serious black magic)

File: octane/handlers/upgrade/compute.py [hunk 5]
```diff
@@ -63,3 +66,12 @@ def shutoff_vms(self):
                '|',
                'xargs -I% nova stop %']
         out, err = ssh.call(cmd, stdout=ssh.PIPE, node=self.node)
+
+    def create_extra_partition(self):
```
Comment (ogelbukh) [inline octane/handlers/upgrade/compute.py]: I would call this method 'create_configdrive_partition', but it is optional.

Comment (SlavaValAl) [inline octane/handlers/upgrade/compute.py]: fixed

File: octane/patches/update_node_partition_info.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        else:
+            if v['type'] == 'lvm_meta_pool' or v['type'] == 'boot':
+                v['size'] = 0
+            disk['volumes'].append(v)
+nv.volumes.append({})
+db.flush()
+db.commit()
+nv.volumes = nv.volumes[:-1]
+db.flush()
+db.commit()
```
Comment (YorikSar) [inline octane/patches/update_node_partition_info.py]: Why do we add and remove this last item? Please remove code or add comment about why we're doing this.

Comment (SlavaValAl) [inline octane/patches/update_node_partition_info.py]: will fix it

Comment (SlavaValAl) [inline octane/patches/update_node_partition_info.py]: fixed

**Pull request merged**